### ISSUE TITLE
Re-adds 95 domains at every run that is already in the pihole

### ISSUE THREAD

**Issue by SireSalty** [anudeepND/whitelist#246]
Hello, not using docker just the python script.    The script appears to work but it seems to not be extracting gravity properly to check that the items are previously added, as it seems to re-add every time its run.  Raspberry pi 3, raspbian bullseye 9-6-22 update and pihole PI-HOLE [V5.12](https://github.com/pi-hole/pi-hole/releases/v5.12) FTL [V5.17](https://github.com/pi-hole/FTL/releases/v5.17) WEB INTERFACE [V5.14.2](https://github.com/pi-hole/AdminLTE/releases/v5.14.2).
---------------script run----------------------
This script will download and add domains from the repo to whitelist.
All the domains in this list are safe to add and does not contain any tracking or adserving domains.

[i] Pi-hole path exists
[i] Write access to /etc/pihole verified
[i] Pi-Hole Gravity database found
[i] 191 domains and 191 SQL queries discovered
[i] Connecting to Gravity.
[i] Successfully Connected to Gravity.
[i] Checking Gravity for domains added by script.
[i] 96 Domains from script in whitelist already.
[i] Checking Gravity for domains added by user that are also in script.
[i] Found 0 domains added by the user that would be added by script.
[i] Checking Gravity for domains previously added by script that are NOT in new script.
[i] Found 0 domain(s) added previously by script that are not in script.
[i] Checking script for domains not in Gravity.
[i] 95 domain(s) NOT in Gravity that are in new script.

    - Adding android.clients.google.com
    - Adding appleid.apple.com
    - Adding apps.skype.com
    - Adding apt.sonarr.tv
    - Adding attestation.xboxlive.com
    - Adding c.s-microsoft.com
    - Adding cdn.vidible.tv
    - Adding cert.mgt.xboxlive.com
    - Adding clientconfig.passport.net
    - Adding clients2.google.com
    - Adding clients3.google.com
    - Adding clients4.google.com
    - Adding cpms.spop10.ams.plex.bz
    - Adding cpms35.spop10.ams.plex.bz
    - Adding ctldl.windowsupdate.com
    - Adding livepassdl.conviva.com
    - Adding gfwsl.geforce.com
    - Adding appspot-preview.l.google.com
    - Adding dashboard.plex.tv
    - Adding def-vef.xboxlive.com
    - Adding delivery.vidible.tv
    - Adding dev.virtualearth.net
    - Adding device.auth.xboxlive.com
    - Adding displaycatalog.mp.microsoft.com
    - Adding dl.delivery.mp.microsoft.com
    - Adding dl.dropboxusercontent.com
    - Adding download.sonarr.tv
    - Adding ecn.dev.virtualearth.net
    - Adding edge.api.brightcove.com
    - Adding eds.xboxlive.com
    - Adding forums.sonarr.tv
    - Adding g.live.com
    - Adding geo-prod.do.dsp.mp.microsoft.com
    - Adding gravatar.com
    - Adding help.ui.xboxlive.com
    - Adding imagesak.secureserver.net
    - Adding img.vidible.tv
    - Adding licensing.xboxlive.com
    - Adding login.live.com
    - Adding login.microsoftonline.com
    - Adding meta-db-worker02.pop.ric.plex.bz
    - Adding meta.plex.bz
    - Adding meta.plex.tv
    - Adding msftncsi.com
    - Adding nine.plugins.plexapp.com
    - Adding node.plexapp.com
    - Adding notify.xboxlive.com
    - Adding ns1.dropbox.com
    - Adding ns2.dropbox.com
    - Adding o1.email.plex.tv
    - Adding o2.sg0.plex.tv
    - Adding officeclient.microsoft.com
    - Adding outlook.office365.com
    - Adding placehold.it
    - Adding placeholdit.imgix.net
    - Adding pricelist.skype.com
    - Adding products.office.com
    - Adding proxy.plex.bz
    - Adding proxy.plex.tv
    - Adding proxy02.pop.ord.plex.bz
    - Adding pubsub.plex.bz
    - Adding pubsub.plex.tv
    - Adding raw.githubusercontent.com
    - Adding redirector.googlevideo.com
    - Adding s.gateway.messenger.live.com
    - Adding s.marketwatch.com
    - Adding s.youtube.com
    - Adding sa.symcb.com
    - Adding services.sonarr.tv
    - Adding skyhook.sonarr.tv
    - Adding spclient.wg.spotify.com
    - Adding staging.plex.tv
    - Adding status.plex.tv
    - Adding t0.ssl.ak.dynamic.tiles.virtualearth.net
    - Adding t0.ssl.ak.tiles.virtualearth.net
    - Adding themoviedb.com
    - Adding thetvdb.com
    - Adding tinyurl.com
    - Adding title.auth.xboxlive.com
    - Adding title.mgt.xboxlive.com
    - Adding tvdb2.plex.tv
    - Adding tvthemes.plexapp.com
    - Adding ui.skype.com
    - Adding video-stats.l.google.com
    - Adding videos.vidible.tv
    - Adding widget-cdn.rpxnow.com
    - Adding www.googleapis.com
    - Adding www.msftncsi.com
    - Adding xbox.ipv6.microsoft.com
    - Adding xboxexperiencesprod.experimentation.xboxlive.com
    - Adding xflight.xboxlive.com
    - Adding xkms.xboxlive.com
    - Adding xsts.auth.xboxlive.com
    - Adding connectivitycheck.gstatic.com
    - Adding connectivitycheck.android.com

[i] Checking Gravity for newly added domains.

[i] All 95 new domain(s) have not been added to Gravity.
[i] There are a total of 468 domains in your whitelist (regex(6) & exact(462))
[i] The database connection is closed
[i] Restarting Pi-hole. This could take a few seconds

Done. Happy ad-blocking :)

Star me on GitHub: https://github.com/anudeepND/whitelist
Buy me a coffee: https://paypal.me/anudeepND
---------------run end----------------

**Comment by Technetium1** [comment]
Have you made any modifications to anything?

**Comment by SireSalty** [comment]
> Technetium1

no, this is a fresh pi install and the only thing I did was import the teleporter from the old one.
